### DISCUSSION TITLE
Enable sending open telemetry data to a custom backend

### DISCUSSION THREAD

**Discussion by david-gang** [PrefectHQ/prefect#16068] [Ideas]
I see currently in the code in prefect 3.13 that it is just possible to configure if to send telemetry to prefect cloud or not

```
    telemetry_enabled: bool = Field(
        default=False,
        description="Enables sending telemetry to Prefect Cloud.",
    )
```

I have my own otel backend and I am interested to see how my code behaves. I want the possibility to send the traces (and logs in the future) to my backend. I would like a way which enables us to send the otel data to a custom backend.

**Comment by bunchesofdonald** [comment]
Hey @david-gang! We're currently instrumenting the engines and some server-side events and are working on getting that all to Prefect Cloud. But our _eventual_ goal is to allow sending of telemetry anywhere.

In _theory_ you could begin doing that now, though I'm not sure of the quality of the instrumentation you'd be getting as it's still being implemented. There's no documentation right now, but you could look at the cloud setup to get an idea of what you might need on your side: https://github.com/PrefectHQ/prefect/blob/main/src/prefect/telemetry/instrumentation.py#L82-L93

Let us know if you get it up and running and how those traces are looking!

**Comment by jbw-vtl** [comment]
Also wanted to express interest here, we actively use OTEL > DataDog, having the option to hook into the existing otel collectors would be amazing

Keen on future developments here
